sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -216,16 +276,37 @@ public String getId() {
         if (options == null) {
             options = new CosmosItemRequestOptions();
         }
+
+        final CosmosItemRequestOptions requestOptions = options;
+        return withContext(context -> createItem(item, requestOptions, context)).subscriberContext(reactorContext -> {
+            Optional<String> master = reactorContext.getOrEmpty(TracerProvider.MASTER_CALL);
+            if (master.isPresent()) {
+                reactorContext = reactorContext.put(TracerProvider.NESTED_CALL, true);
+            }
+            return reactorContext.put(TracerProvider.MASTER_CALL, true);
```

Comment:
exactly the same code, can this be factored out

```java
return withContext(context -> replace(containerProperties, requestOptions, context))
    .subscriberContext(reactorContext -> factoredOutFunc())
```

Reply:
Done